client/flutter/lib/pages/main_pages/app_tab_router.dart

```diff
@@ -40,6 +41,52 @@ class AppTabRouter extends StatelessWidget {
 
   AppTabRouter(this.tabs, this.navItems);
 
+  @override
+  _AppTabRouterState createState() => _AppTabRouterState(tabs, navItems);
+}
+
+class _AppTabRouterState extends State<AppTabRouter> {
+  CupertinoTabController _controller;
+  FirebaseAnalytics _analytics;
+  int lastTabIndex;
+  @override
+  void initState() {
+    _analytics = FirebaseAnalytics();
+    _controller = CupertinoTabController();
+    lastTabIndex = _controller.index;
+
+    _controller.addListener(
+      () {
+        int currentIndex = _controller.index;
+
+        String lastTab = tabTitle(lastTabIndex);
+        String currentTab = tabTitle(currentIndex);
+
+        lastTabIndex = currentIndex;
+
+        _analytics.logEvent(
+          name: 'changed_tabs',
+          parameters: {
+            'current_tab': currentTab,
+            'last_tab': lastTab,
+          },
+        );
+      },
+    );
+    super.initState();
+  }
+
+  String tabTitle(int index) {
+    Text txt = AppTabRouter.defaultNavItems[index].title;
+    String data = txt.data;
+    return data;
+  }
+
+  final List<Widget Function(BuildContext)> tabs;
+  final List<BottomNavigationBarItem> navItems;
+
+  _AppTabRouterState(this.tabs, this.navItems);
```

Comment:
Same comment as above - access these vars through the widget, if they're needed at all

```suggestion
  const _AppTabRouterState();
```